src/Conversion/KrnlToLLVM/KrnlToLLVM.cpp

```diff
@@ -327,9 +327,15 @@ class KrnlGlobalOpLowering : public ConvertToLLVMPattern {
     auto constantElementType =
         typeConverter.convertType(memRefTy.getElementType());
     auto globalType = constantElementType;
-    for (int i = shape.size() - 1; i >= 0; i--)
-      globalType = LLVM::LLVMType::getArrayTy(
-          globalType.cast<LLVM::LLVMType>(), ArrayAttrIntVal(shape, i));
+
+    if (shape.empty()) {
```

Comment:
@doru1004 this is the necessary modification to get LLVM-level translation to work. Basically, LLVM global op lowering thinks that the value attribute with type tensor<i64> is a splat elements attribute, which is true, but still somewhat surprising (and I don't know whether not treating a scalar tensor as a splat element would solve this issue more elegantly, but if that is the case then it would be something we can contribute to MLIR upstream). And to lower a splat element attribute, the attribute needs to have either array type or vector type whereas the scalar type is not supported. To workaround this issue I convert the scalar to become an LLVM array of size 1. This seems fine because the shape information of the lowered LLVM array is never used to re-construct the memref anyways.

c.f. https://github.com/llvm/llvm-project/blob/047400ed8204ebcc0b361ca9285b34ea91479b69/mlir/lib/Target/LLVMIR/ModuleTranslation.cpp#L124

Reply:
I think the change makes sense :)